Feature/payhere 4 API 리팩토링

GoHeeSeok00 opened the pull request:
### Types of changes

- [ ] 기능 추가
- [ ] 버그 수정
- [ ] 테스트케이스
- [ ] 컨벤션 수정
- [ ] 문서
- [ ] CI/CD
- [x] 리팩토링

### Summary

- API 모델 구조, 시리얼라이저, 메서드, url 리팩토링 진행

### Changes

 - date 및 deleted_at 필드 추가로 인한 구조 수정        
 - path 파라미터의 obj_id를 각각의 테이블 명의 id로 수정
 - 삭제된 기록을 조회하기 위한 시리얼라이저 추가        
 - put안에 있는 3개의 행위 분리                         
    - 1class put 수정                                   
    - 1class patch 소프트 삭제                          
    - 2class patch 복구" 

### 특이사항 (Optional)

-

### Screenshots (Optional)

GoHeeSeok00 committed:
Update : AccountBook, AccountBookRecord 필드 추가

- deleted_at : 삭제 시간을 저장하기 위한 필드 추가 (AccountBook,AccountBookRecord)
- date : 날짜 지정을 위한 필드 추가 (AccountBookRecord)

GoHeeSeok00 committed:
Merge branch 'main' of https://github.com/pre-onboarding-3rd-team-H/02_Payhere_TeamH into feature/PAYHERE-4

GoHeeSeok00 committed:
Refactor : api 리팩토링 진행

- date 및 deleted_at 필드 추가로 인한 구조 수정
- path 파라미터의 obj_id를 각각의 테이블 명의 id로 수정
- 삭제된 기록을 조회하기 위한 시리얼라이저 추가
- put안에 있는 3개의 행위 분리
   - 1class put 수정
   - 1class patch 소프트 삭제
   - 2class patch 복구

tkdqor commented on account_book/views.py:
```diff
@@ -93,21 +89,26 @@ def get_object_and_check_permission(self, obj_id):
         return object
 
-    def get(self, request, obj_id):
+    def get(self, request):
```
해당 get 메서드가 AccountBooksRecordAPIView클래스 내부에 있으니까 가계부 단건 조회를 하려면 obj_id가 필요하지 않을까요?

GoHeeSeok00 replied:
넵 api 테스트 시 문제점 발견하여 수정, 푸시 했습니다!!

tkdqor commented on account_book/views.py:
```diff
@@ -93,21 +89,26 @@ def get_object_and_check_permission(self, obj_id):
 
-        클라이언트의 요청 및 가계부 기록 고유번호 입력 시, 특정 기록에 대한 내용을 응답하는 메서드입니다.
-        설정한 permission_classes로 인해 특정 기록의 작성자가 아닐 경우, 접근이 제한됩니다.
+        쿼리 파라미터로 "status" 키의 값이 "delete"가 들어오는 경우 삭제된 데이터를 보여줍니다.
```
위와 마찬가지로 AccountBooksRecordAPIView클래스 내부인데 쿼리 파라미터가 안 붙는 거 아닌가 싶습니다.

GoHeeSeok00 replied:
![image](https://user-images.githubusercontent.com/96563183/177913925-aa113ea1-e8cc-4b19-804e-343eddb54b7e.png)

쿼리 파라미터는 이런형태로 붙게됩니다!

tkdqor replied:
record 단건에 대한 삭제 내역이군요! 제가 제대로 확인을 못했네요 캡처까지 감사합니다:)

GoHeeSeok00 committed:
Fix : api 테스트 시 발생 오류 수정

- 수정 진행 시 is_deleted 가 none이 아닐때 잘못된 요청 처리
- 단일 가계부의 삭제 내역을 조회할 때 잘못된 모델 참조 수정
- 주석 및 예외처리 추가

nmdkims reviewed (approved):
Date 관련 수정사항 확인하였습니다. 해당 내용에 맞게 테스트 케이스 구성하도록 하겠습니다.

GoHeeSeok00 committed:
Fix : 충돌 수정
